frontend/src/components/home/measurements/client/ClientsAndProjectsEvolution.tsx

```diff
@@ -0,0 +1,142 @@
+import { Row } from "react-bootstrap";
+import ClientsAndProjectsEvolutionChart from "./ClientsAndProjectsEvolutionChart";
+import ClientsAndProjectsEvolutionList from "./ClientsAndProjectsEvolutionList";
+import { Container } from "react-bootstrap";
+import { Col } from "react-bootstrap";
+import { useState } from "react";
+import moment, { Moment } from "moment";
+import { useSelector } from "react-redux";
+import { ConfacState } from "../../../../reducers/app-state";
+import { t } from "../../../utils";
+import { Link } from "react-router-dom";
+import PeriodPicker from "../PeriodPicker";
+
+export interface clientChartData {
+  year: number;
+  monthIndex: number;
+  month: string;
+  clients: number | null;
+  clientsWithProjects: number | null;
+}
+const ClientsAndProjectsEvolution = () => {
+  const [from, setFrom] = useState<moment.Moment>(moment());
+  const [to, setTo] = useState<moment.Moment>(from.clone().add(11, "month"));
+  const models = useSelector((state: ConfacState) => ({
+    clients: state.clients,
+    projects: state.projects,
+  }));
+
+  let interim = from.clone();
+  let timeValues: Moment[] = [];
+
+  while (interim.startOf("month").isSameOrBefore(to.endOf("month"))) {
+    timeValues.push(interim.clone());
+    interim.add(1, "month");
+  }
+
+  const dataSet: clientChartData[] = timeValues.map((date) => {
+    return {
+      year: date.year(),
+      monthIndex: date.month(),
+      month: date.format("MMM"),
+      clients: null,
+      clientsWithProjects: null,
+    };
+  });
+
+  const clientsWithProject = models.clients.map((client) => {
+    return {
+      detail: client,
+      projects: models.projects.filter(
+        (project) => project.client.clientId === client._id
+      ),
+    };
+  });
+
+  let count: number = 0;
+  dataSet.forEach((entry) => {
+    let amount: number = models.clients
+      .filter(
+        (client) => moment.utc(client.audit.createdOn).year() === entry.year
+      )
+      .filter(
+        (client) =>
+          moment.utc(client.audit.createdOn).month() === entry.monthIndex
+      ).length;
```

Comment:
Ugh, pas dit aan in de style zoals andere files.
Dit is verschrikkelijk onleesbaar.

```
.filter(
        (client) =>
          moment.utc(client.audit.createdOn).month() === entry.monthIndex
      )
```

4 lijnen code...

```
.filter(client => moment.utc(client.audit.createdOn).month() === entry.monthIndex)
```